mne/utils/mixin.py

```diff
@@ -448,9 +473,55 @@ def tmax(self):
         """Last time point."""
         return self.times[-1]
 
+    @verbose
+    def crop(self, tmin=None, tmax=None, include_tmax=True, verbose=None):
+        """Crop data to a given time interval.
+
+        Parameters
+        ----------
+        tmin : float | None
+            Start time of selection in seconds.
+        tmax : float | None
+            End time of selection in seconds.
+        %(include_tmax)s
+        %(verbose)s
+
+        Returns
+        -------
+        inst : MNE-object
+            The cropped time-series object, modified in-place.
+
+        Notes
+        -----
+        %(notes_tmax_included_by_default)s
+        """
+        if tmin is None:
+            tmin = self.tmin
+        elif tmin < self.tmin:
+            warn(f'tmin is not in time interval. tmin is set to '
+                 f'inst.tmin ({self.tmin:g} sec)')
```

Comment:
for these warnings, can we figure out from `self` what the class is, and insert the string `Epochs` or `Evoked` or whatever into the warning, in place of `inst`?

Reply:
Sure